service/snap.go

```diff
@@ -175,31 +174,41 @@ func (s *service) UnlinkTargets(ctx context.Context, symID, srcDevID string, pma
 	return nil
 }
 
-// IsSnapshotLicensed return true if the symmetrix array has
-// SnapVX license
+// This function checks if the PowerMax array has the SnapVX license.
+// It returns an error if the array does not meet the expectations.
 func (s *service) IsSnapshotLicensed(ctx context.Context, symID string, pmaxClient pmax.Pmax) (err error) {
 	if _, err := pmaxClient.IsAllowedArray(symID); err != nil {
 		return err
 	}
 	mutex.Lock()
 	defer mutex.Unlock()
-	if licenseCached == false {
-		symmRepCapabilities, err = pmaxClient.GetReplicationCapabilities(ctx)
+
+	symmRepCapability, ok := symmRepCapabilities[symID]
+	if !ok {
+		repCapabilities, err := pmaxClient.GetReplicationCapabilities(ctx)
 		if err != nil {
 			return err
 		}
-		licenseCached = true
-		log.Infof("License information with Powermax %s is cached", symID)
-	}
-	for i := range symmRepCapabilities.SymmetrixCapability {
-		if symmRepCapabilities.SymmetrixCapability[i].SymmetrixID == symID {
-			if symmRepCapabilities.SymmetrixCapability[i].SnapVxCapable {
-				return nil
+
+		for _, symmCapability := range repCapabilities.SymmetrixCapability {
+			if symmCapability.SymmetrixID == symID {
+				symmRepCapabilities[symID] = symmCapability
+				log.Infof("License information with PowerMax %s is cached", symID)
```

Comment:
The log message does not seem to reflect what the code is doing. Someone in service may see this and not know we are checking a license.